core/domain/question_fetchers.py

```diff
@@ -63,14 +79,21 @@ def get_questions_and_skill_descriptions_by_skill_ids(
 
     for skill_ids_list in grouped_skill_ids:
         skills = skill_models.SkillModel.get_multi(skill_ids_list)
-        grouped_skill_descriptions.append(
-            [skill.description if skill else None for skill in skills])
+        skill_descriptions: List[Optional[str]] = []
+        for skill in skills:
+            if skill:
+                # Ruling out the possibility of None for mypy type checking.
+                assert isinstance(skill.description, str)
+                skill_descriptions.append(skill.description)
+        grouped_skill_descriptions.append(skill_descriptions)
```

Comment:
I don't think we have preserved the case where skill is `None` i.e
`grouped_skill_descriptions.append( [skill.description if skill else None for skill in skills])`
Can we either preserve the above statement or add a test if we are breaking it into different lines?

Reply:
Done